sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/DatabaseTestHelper.java

```diff
@@ -100,13 +100,21 @@ public static void createTable(
             .withMaxRetries(4)
             .backoff();
     while (true) {
-      try (Connection connection = dataSource.getConnection()) {
+      //This is not implemented as try-with-resources because it appears that try-with-resources is
+      //not correctly catching the PSQLException thrown by dataSource.getConnection()
+      Connection connection = null;
+      try {
+        connection = dataSource.getConnection();
         try (Statement statement = connection.createStatement()) {
           statement.execute(String.format("create table %s (%s)", tableName, fieldsList));
           return;
         }
       } catch (SQLException e) {
         exception = e;
+      } finally {
+        if (connection!=null){
+          connection.close();
```

Comment:
Probably it is closing the connecting caused issue. dataSource owns a connection pool and manage the connections. Closing it pre-maturely may have caused racing conditions.

Reply:
it looks like it is happening in .getConnection based on the stacktrace